core/core_bind.cpp

```diff
@@ -1072,7 +1072,9 @@ Error File::open_compressed(const String &p_path, ModeFlags p_mode_flags, Compre
 }
 
 Error File::open(const String &p_path, ModeFlags p_mode_flags) {
-	close();
+	if (f.is_valid()) {
+		close();
+	}
```

Comment:
This can be removed at all, since setting `f` will unref and close a previously open file automatically.

Reply:
Done